src/optimizer/query_rewrite.c

```diff
@@ -3074,6 +3088,410 @@ qo_reduce_comp_pair_terms (PARSER_CONTEXT * parser, PT_NODE ** wherep)
+
+      /* col1(my spec) = const */
+      if (info1.my_spec_cnt == 1 && info1.other_spec_cnt == 0)
+	{
+	  /* const */
+	  if (info2.my_spec_cnt == 0)
+	    {
+	      point_list = parser_append_node (pt_point (parser, info1.my_spec_node), point_list);
+	    }
+	}
+      /* const = col(my spec) */
+      else if (info2.my_spec_cnt == 1 && info2.other_spec_cnt == 0)
+	{
+	  /* const */
+	  if (info1.my_spec_cnt == 0)
+	    {
+	      point_list = parser_append_node (pt_point (parser, info2.my_spec_node), point_list);
+	    }
+	}
+      pred = pred->next;
+    }
+  return point_list;
+}
+
+/*
+ * qo_modify_location () -
+ *   return:
+ *   parser(in):
+ *   node(in):
+ *   arg(in):
+ *   continue_walk(in):
+ */
+static PT_NODE *
+qo_modify_location (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *continue_walk)
+{
+  RESET_LOCATION_INFO *infop = (RESET_LOCATION_INFO *) arg;
+
+  if (node->node_type == PT_EXPR && node->info.expr.location == infop->start)
+    {
+      node->info.expr.location = infop->end;
+    }
+
+  if (node->node_type == PT_NAME && node->info.name.location == infop->start)
+    {
+      node->info.name.location = infop->end;
+    }
+
+  if (node->node_type == PT_VALUE && node->info.value.location == infop->start)
+    {
+      node->info.value.location = infop->end;
+    }
+
+  return node;
+}
+
+/*
+ * qo_reset_spec_location () - reset location of spec
+ *   return: node_list
+ *   parser(in):
+ *   on_cond(in):
+ *   spec(in):
+ *
+ * Note:
+ */
+static void
+qo_reset_spec_location (PARSER_CONTEXT * parser, PT_NODE * spec, PT_NODE * query)
+{
+  short curr_loc, after_loc;
+  PT_NODE *where;
+  RESET_LOCATION_INFO locate_info;
+
+  while (spec != NULL)
+    {
+      curr_loc = spec->info.spec.location;
+      after_loc = curr_loc - 1;
+
+      if (curr_loc <= 0 || after_loc <= 0)
+	{
+	  spec = spec->next;
+	  continue;
+	}
+
+      /* reset location of spec */
+      spec->info.spec.location = after_loc;
+
+      /* reset location of predicate */
+      locate_info.start = curr_loc;
+      locate_info.end = after_loc;
+      where = query->info.query.q.select.where;
+      (void) parser_walk_tree (parser, where, qo_modify_location, &locate_info, NULL, NULL);
+
+      spec = spec->next;
+    }
+}
+
+/*
+ * qo_reduce_outer_joined_tables () - reduce outer joined tables with unique join predicates
+ *   return:
+ *   parser(in):
+ *   spec(in):
+ *   query(in):
+ *
+ * Note:
+ * 	examples:
+ *  	          select count(*)
+ *                  from tbl1 a left outer join tbl2 b on a.pk = b.pk <== outer joined table with unique join columns
+ *                ==>
+ *                select count(*)
+ *                  from tbl1 a
+ */
+static PT_NODE *
+qo_reduce_outer_joined_tables (PARSER_CONTEXT * parser, PT_NODE * spec, PT_NODE * query)
+{
+  MOP cls;
+  SM_CLASS_CONSTRAINT *consp;
+  SM_ATTRIBUTE *attrp;
+  PT_NODE *point_list, *point, *where, *col, *tmp_spec, *prev_spec, *pred, *prev_pred, *next_pred;
```

Comment:
```suggestion
  PT_NODE *point_list = NULL, *point, *where, *col, *tmp_spec, *prev_spec, *pred, *prev_pred, *next_pred;
```

at the `end` flag, point_list is freed according to the case this variable is NULL.

Reply:
I'll fix it.